Mathlib/RingTheory/Ideal/QuotientOperations.lean

```diff
@@ -670,4 +670,194 @@ theorem quotQuotEquivComm_algebraMap (x : R) :
+
+@[simp]
+-- Porting note: had to add an extra coercion arrow on the right hand side.
+theorem coe_quotQuotEquivQuotSupₐ : ⇑(quotQuotEquivQuotSupₐ R I J) = ⇑(quotQuotEquivQuotSup I J) :=
+  rfl
+#align double_quot.coe_quot_quot_equiv_quot_supₐ DoubleQuot.coe_quotQuotEquivQuotSupₐ
+
+@[simp]
+theorem quotQuotEquivQuotSupₐ_symm_toRingEquiv :
+    ((quotQuotEquivQuotSupₐ R I J).symm : _ ≃+* _ ⧸ J.map (Quotient.mkₐ R I)) =
+      (quotQuotEquivQuotSup I J).symm :=
+  rfl
+#align double_quot.quot_quot_equiv_quot_supₐ_symm_to_ring_equiv DoubleQuot.quotQuotEquivQuotSupₐ_symm_toRingEquiv
+
+@[simp]
+-- Porting note: had to add an extra coercion arrow on the right hand side.
+theorem coe_quotQuotEquivQuotSupₐ_symm :
+    ⇑(quotQuotEquivQuotSupₐ R I J).symm = ⇑(quotQuotEquivQuotSup I J).symm :=
+  rfl
+#align double_quot.coe_quot_quot_equiv_quot_supₐ_symm DoubleQuot.coe_quotQuotEquivQuotSupₐ_symm
+
+/-- The natural algebra isomorphism `(A / I) / J' → (A / J) / I'`,
+  where `J'` (resp. `I'`) is the projection of `J` in `A / I` (resp. `I` in `A / J`). -/
+def quotQuotEquivCommₐ :
+    ((A ⧸ I) ⧸ J.map (Quotient.mkₐ R I)) ≃ₐ[R] (A ⧸ J) ⧸ I.map (Quotient.mkₐ R J) :=
+  @AlgEquiv.ofRingEquiv R _ _ _ _ _ _ _ (quotQuotEquivComm I J) fun _ => rfl
+#align double_quot.quot_quot_equiv_commₐ DoubleQuot.quotQuotEquivCommₐ
+
+@[simp]
+theorem quotQuotEquivCommₐ_toRingEquiv :
+    (quotQuotEquivCommₐ R I J : _ ⧸ J.map (Quotient.mkₐ R I) ≃+* _ ⧸ I.map (Quotient.mkₐ R J)) =
+      quotQuotEquivComm I J :=
+  -- Porting note: should just be `rfl` but `AlgEquiv.toRingEquiv` and `AlgEquiv.ofRingEquiv`
+  -- involve repacking everything in the structure, so Lean ends up unfolding `quotQuotEquivComm`
+  -- and timing out.
+  RingEquiv.ext fun _ => rfl
+#align double_quot.quot_quot_equiv_commₐ_to_ring_equiv DoubleQuot.quotQuotEquivCommₐ_toRingEquiv
+
+@[simp]
+theorem coe_quotQuotEquivCommₐ : ⇑(quotQuotEquivCommₐ R I J) = ⇑(quotQuotEquivComm I J) :=
+  rfl
+#align double_quot.coe_quot_quot_equiv_commₐ DoubleQuot.coe_quotQuotEquivCommₐ
+
+@[simp]
+theorem quotQuotEquivComm_symmₐ : (quotQuotEquivCommₐ R I J).symm = quotQuotEquivCommₐ R J I := by
+  -- Porting note: should just be `rfl` but `AlgEquiv.toRingEquiv` and `AlgEquiv.ofRingEquiv`
+  -- involve repacking everything in the structure, so Lean ends up unfolding `quotQuotEquivComm`
+  -- and timing out.
+  ext
+  unfold quotQuotEquivCommₐ
+  congr
+#align double_quot.quot_quot_equiv_comm_symmₐ DoubleQuot.quotQuotEquivComm_symmₐ
+
+@[simp]
+theorem quotQuotEquivComm_comp_quotQuotMkₐ :
+    AlgHom.comp (↑(quotQuotEquivCommₐ R I J)) (quotQuotMkₐ R I J) = quotQuotMkₐ R J I :=
+  AlgHom.ext <| quotQuotEquivComm_quotQuotMk I J
+#align double_quot.quot_quot_equiv_comm_comp_quot_quot_mkₐ DoubleQuot.quotQuotEquivComm_comp_quotQuotMkₐ
+
+variable {I J}
+
+/-- The **third isomorphism theorem** for rings. See `quotQuotEquivQuotSupₐ` for version
```

Comment:
"Rings" should read "algebras". This is carried over from an upstream error I made in Lean 3 mathlib.